### DISCUSSION TITLE
check out the implementation of ppl with strided sliding window strategy

### DISCUSSION THREAD

**Discussion by qiufengyuyi** [open-compass/opencompass#216] [View all discussions]
hi,i check the implementation of ppl for llm from https://huggingface.co/docs/transformers/perplexity. it uses a strided sliding window strategy to calculate ppl. But in opencompass, no windows are applied. Is that because you set de windows size as the max length, and all the input size is no more than that length?

**Comment by Ezra-Yu** [comment]
Yes, we do not use windows, we use the tire inputs.

**Reply by qiufengyuyi** [reply]
is there any problem when not using windows?

**Reply by Ezra-Yu** [reply]
Our way is to construct a complete sentence and calculate its PPL. For example, "Question: xxx A) yyyy, B) zzzzz, The Answer is A". Calculate the PPL of this input, not the one mentioned in huggingface document which calculates it while completing.

**Reply by qiufengyuyi** [reply]
so,when the task is not like choice-answering ,but more like a text generation(like text summarization),we should apply windows instead. am i right ?

**Reply by tonysy** [reply]
Would you like to give an example on how to evaluate a generation task with PPL?

**Reply by qiufengyuyi** [reply]
![image](https://github.com/InternLM/opencompass/assets/19869291/7e8885a1-7669-48ba-92e8-b17cbcc578d5)

examples from huggingface ,or  writing a fiction/poet/story when there's no completely "right" reference

**Reply by tonysy** [reply]
So you just want to calculate the PPL without 'real' answer matching process?

**Reply by qiufengyuyi** [reply]
yes, there should be some conditions that we do not have answers ,just need to evaluate the quality of a generation from a llm , but maybe the ppl is not the best choice for this test. just check the implementation of ppl when analyzing your code and examples from huggingface.

**Reply by tonysy** [reply]
Get it. The current implementation is designed for evaluation the Multi-Choice-Quesitons, we will add this feature into our backlog. If you are interested, welcome to contribute such an implementation into OpenCompass.
